core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -314,22 +336,45 @@ private void updateTopicOverviewItems(
     }
   }
 
-  private boolean isRequestAlreadyOpen(String topicNameSearch, String environmentId, int tenantId) {
-
-    List<TopicRequest> topicReqs =
-        manageDatabase
-            .getHandleDbRequests()
-            .getAllTopicRequests(
-                getUserName(),
-                RequestStatus.CREATED.value,
-                null,
-                environmentId,
-                topicNameSearch,
-                false,
-                tenantId);
-    log.debug("Open Reqs for Topics {} , size {}", topicReqs, topicReqs.size());
-    // there should only ever be 1 delete request and in any other scenario this should be 0.
-    return topicReqs.size() >= 1;
+  private boolean isTopicRequestOpen(String topicName, String environmentId, int tenantId) {
+    return manageDatabase
+        .getHandleDbRequests()
+        .existsTopicRequest(topicName, RequestStatus.CREATED.value, environmentId, tenantId);
+  }
+
+  private boolean isTopicPromoteRequestOpen(String topicName, String environmentId, int tenantId) {
+    return manageDatabase
+        .getHandleDbRequests()
+        .existsTopicRequest(
+            topicName,
+            RequestStatus.CREATED.value,
+            RequestOperationType.PROMOTE.value,
+            environmentId,
+            tenantId);
+  }
+
+  private boolean isACLRequestOpen(String topicName, String environmentId, int tenantId) {
+
+    return manageDatabase
+        .getHandleDbRequests()
+        .existsAclRequest(topicName, RequestStatus.CREATED.value, environmentId, tenantId);
+  }
+
+  private boolean isSchemaRequestOpen(String topicName, String envId, int tenantId) {
+    return manageDatabase
+        .getHandleDbRequests()
+        .existsSchemaRequest(topicName, RequestStatus.CREATED.value, envId, tenantId);
+  }
+
+  private boolean isSchemaPromoteRequestOpen(String topicName, String envId, int tenantId) {
```

Comment:
this method also seems to be unused

Reply:
This is for the next PR where we add a check to see if a schema promote request is already created. As I was doing the others I wanted to add this in at the same time. It makes the next change smaller.